### PR TITLE
[Hetero plugin] Model caching support

### PR DESCRIPTION
**Pull request by nosovmik**
### Details:
- Enable IMPORT_EXPORT_SUPPORT metric, add DEVICE_ARCHITECTURE metric support
- Introduce internal FORCE_DISABLE_CACHE flag to avoid caching of subnetworks
- Added test for hetero with plugins which don't support caching
- Added test for hetero with plugins supported different cache architecture

### Tickets:
 - 51877
 - 49318

### CODE REVIEW

File: inference-engine/src/hetero_plugin/hetero_plugin.cpp [hunk 1]
```diff
@@ -164,7 +165,8 @@ Parameter Engine::GetMetric(const std::string& name, const std::map<std::string,
         IE_SET_METRIC_RETURN(SUPPORTED_METRICS, std::vector<std::string>{
             METRIC_KEY(SUPPORTED_METRICS),
             METRIC_KEY(FULL_DEVICE_NAME),
-            METRIC_KEY(SUPPORTED_CONFIG_KEYS)});
+            METRIC_KEY(SUPPORTED_CONFIG_KEYS),
```
Comment (ilya-lavrenov) [inline inference-engine/src/hetero_plugin/hetero_plugin.cpp:168]: I think we also need HETERO arch. Is it supposed in a separate PR?

**Review by ilya-lavrenov — APPROVED** [review]

File: inference-engine/src/hetero_plugin/hetero_plugin.cpp [hunk 2]
```diff
@@ -154,23 +155,54 @@ QueryNetworkResult Engine::QueryNetwork(const CNNNetwork &network, const Configs
...
     }
 }
+std::string Engine::DeviceArchitecture(const std::string& targetFallback) const {
+    auto fallbackDevices = InferenceEngine::DeviceIDParser::getHeteroDevices(targetFallback);
+    std::string resArch;
+    for (const auto& device : fallbackDevices) {
+        InferenceEngine::DeviceIDParser parser(device);
+        std::map<std::string, Parameter> options;
+        if (!parser.getDeviceID().empty()) {
+            options.insert({CONFIG_KEY(DEVICE_ID), parser.getDeviceID()});
```
Comment (ilya-lavrenov) [inline inference-engine/src/hetero_plugin/hetero_plugin.cpp:193]: `options` seems to be not used when you call `GetMetric` below..

File: inference-engine/src/hetero_plugin/hetero_plugin.cpp [hunk 3]
```diff
@@ -154,23 +155,54 @@ QueryNetworkResult Engine::QueryNetwork(const CNNNetwork &network, const Configs
...
         IE_SET_METRIC_RETURN(SUPPORTED_CONFIG_KEYS, std::vector<std::string>{
             HETERO_CONFIG_KEY(DUMP_GRAPH_DOT),
             "TARGET_FALLBACK",
             CONFIG_KEY(EXCLUSIVE_ASYNC_REQUESTS)});
     } else if (METRIC_KEY(FULL_DEVICE_NAME) == name) {
         IE_SET_METRIC_RETURN(FULL_DEVICE_NAME, std::string{"HETERO"});
+    } else if (METRIC_KEY(IMPORT_EXPORT_SUPPORT) == name) {
+        IE_SET_METRIC_RETURN(IMPORT_EXPORT_SUPPORT, true);
+    } else if (METRIC_KEY(DEVICE_ARCHITECTURE) == name) {
+        auto deviceIt = options.find("TARGET_FALLBACK");
```
Comment (ilya-lavrenov) [inline inference-engine/src/hetero_plugin/hetero_plugin.cpp:176]: these `options` come from `LoadNetwork` in `ie_core.cpp`. But what if we use the following code:
```
ie::Core core;
core.SetConfig({ { "TARGET_FALLBACK", "CPU" } }, "HETERO");
... 
```
In this case we have value for `TARGET_FALLBACK`, but `options` are still empty.

Comment (nosovmik) [reply inference-engine/src/hetero_plugin/hetero_plugin.cpp:175]: Fixed

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**